crates/typed-store-derive/src/lib.rs

```diff
@@ -0,0 +1,618 @@
+                        }
+                        _ => panic!("Expected string value for rename"),
+                    }
+                } else {
+                    field_name.clone()
+                };
+                if attrs.contains_key(DB_OPTIONS_DEPRECATE) {
+                    deprecated_cfs.push(field_name.clone());
+                }
+
+                return ((field_name, cf_name, type_str), (inner_type, options));
+            } else {
+                panic!("All struct members must be of type DBMap");
+            }
+        }
+        panic!("All struct members must be of type DBMap");
+    });
+
+    let (field_info, inner_types_with_opts): (Vec<_>, Vec<_>) = info.unzip();
+    let (field_names, cf_names, simple_field_type_names): (Vec<_>, Vec<_>, Vec<_>) =
+        field_info.into_iter().multiunzip();
+
+    // Check for homogeneous types
+    if let Some(first) = simple_field_type_names.first() {
+        simple_field_type_names.iter().for_each(|q| {
+            if q != first {
+                panic!("All struct members must be of same type");
+            }
+        })
+    } else {
+        panic!("Cannot derive on empty struct");
+    };
+
+    let (inner_types, options): (Vec<_>, Vec<_>) = inner_types_with_opts.into_iter().unzip();
+
+    ExtractedStructInfo {
+        field_names,
+        cf_names,
+        inner_types,
+        derived_table_options: options,
+        deprecated_cfs,
+    }
+}
+
+/// Extracts the table options override function
+/// The function must take no args and return Options
+fn get_options_override_function(attr: &Attribute) -> syn::Result<String> {
+    if !attr.path().is_ident(DB_OPTIONS_CUSTOM_FUNCTION) {
+        return Err(syn::Error::new_spanned(
+            attr,
+            format!(
+                "Expected function name in format \
+                `#[{DB_OPTIONS_CUSTOM_FUNCTION} = {{function_name}}]`"
+            ),
+        ));
+    }
+    if let Meta::NameValue(meta) = &attr.meta {
+        if let Expr::Lit(expr) = &meta.value {
+            if let Lit::Str(lit_str) = &expr.lit {
+                return Ok(lit_str.value());
+            }
+        }
+    }
+
+    Err(syn::Error::new_spanned(
+        attr,
+        format!(
+            "Expected function name in format \
+            `#[{DB_OPTIONS_CUSTOM_FUNCTION} = {{function_name}}]`"
+        ),
+    ))
+}
+
+fn extract_generics_names(generics: &Generics) -> Vec<Ident> {
+    generics
+        .params
+        .iter()
+        .map(|g| match g {
+            syn::GenericParam::Type(t) => t.ident.clone(),
+            _ => panic!("Unsupported generic type"),
+        })
+        .collect()
+}
+
+struct ExtractedStructInfo {
+    field_names: Vec<Ident>,
+    cf_names: Vec<Ident>,
+    inner_types: Vec<AngleBracketedGenericArguments>,
+    derived_table_options: Vec<GeneralTableOptions>,
+    deprecated_cfs: Vec<Ident>,
+}
+
+#[proc_macro_derive(DBMapUtils, attributes(default_options_override_fn, rename))]
```

Comment:
For the entire `typed-store-derive` crate, I don't think we need it. The table derived from DBMapUtils is not used in Walrus and I don't think we will use it in the future. Let's just remove them.

Reply:
as discussed offline, we will do this in a follow up PR